File: sdks/ruby/lib/featuresignals/client_options.rb
# frozen_string_literal: true

require_relative "eval_context"

module FeatureSignals
  class ClientOptions
    attr_reader :env_key, :base_url, :polling_interval, :streaming,
                :sse_retry, :timeout, :context

    def initialize(
      env_key:,
      base_url: "https://api.featuresignals.com",
      polling_interval: 30,
      streaming: false,
      sse_retry: 5,
      timeout: 10,
      context: EvalContext.new(key: "server")
    )
      raise ArgumentError, "env_key is required" if env_key.nil? || env_key.empty?

      @env_key = env_key
      @base_url = base_url
      @polling_interval = polling_interval
      @streaming = streaming
      @sse_retry = sse_retry
      @timeout = timeout
      @context = context
    end
  end
end

File: sdks/dotnet/src/FeatureSignals/ClientOptions.cs
namespace FeatureSignals;

/// <summary>
/// Configuration options for <see cref="FeatureSignalsClient"/>.
/// </summary>
public sealed class ClientOptions
{
    /// <summary>Environment key identifying the flag environment.</summary>
    public required string EnvKey { get; init; }

    /// <summary>Base URL of the FeatureSignals API.</summary>
    public string BaseUrl { get; init; } = "https://api.featuresignals.com";

    /// <summary>How often to poll for flag updates when not using SSE.</summary>
    public TimeSpan PollingInterval { get; init; } = TimeSpan.FromSeconds(30);

    /// <summary>Enable Server-Sent Events streaming instead of polling.</summary>
    public bool Streaming { get; init; }

    /// <summary>Delay before reconnecting after an SSE connection drop.</summary>
    [Obsolete("SSE reconnection now uses exponential backoff (1s–30s) with jitter. This property is ignored.")]
    public TimeSpan SseRetry { get; init; } = TimeSpan.FromSeconds(5);

    /// <summary>HTTP request timeout.</summary>
    public TimeSpan Timeout { get; init; } = TimeSpan.FromSeconds(10);

    /// <summary>Default evaluation context used for flag fetches.</summary>
    public EvalContext? DefaultContext { get; init; }
}
